### DISCUSSION TITLE
接口请求问题求助

### DISCUSSION THREAD

**Discussion by vincentzhang6230** [UIGF-org/mihoyo-api-collect#64] [Q&A]
我根据文档请求https://api-takumi-record.mihoyo.com/game_record/app/genshin/api/index时,返回-10001错误代码,能帮我看看是那个参数有问题吗?
另外提个建议:可以给接口添加一些请求示例
![20240429150500](https://github.com/UIGF-org/mihoyo-api-collect/assets/59171010/de15c664-3245-4c45-b6e5-e0684ee7ad09)

**Comment by babanbang** [comment]
建议用2.63.1的版本再试一次看看行不行

**Comment by BTMuli** [comment]
> 可以给接口添加一些请求示例

目前 ApiFox 上的 API 处于消极维护状态，如果没有编辑权限的话可以回复或者邮件联系我，我会开放对应权限的（目前我生成的邀请链接都是只读权限的）

**Comment by vincentzhang6230** [comment]
我生成的DS问题,我把**&b={}**删除了,设置为空字符串后请求,遇到错误码1034了,请求遇到验证码怎么办,我看文档好像没有解决方法

**Comment by BTMuli** [comment]
> 我生成的DS问题,我把**&b={}**删除了,设置为空字符串后请求,遇到错误码1034了,请求遇到验证码怎么办,我看文档好像没有解决方法

目前对于1034的解决方式是请求的时候携带合法的deviceFp，虽然文档可能没有但是UIGF成员的相关项目应该不难找到对应issue

**Comment by TRIM5829** [comment]
https://github.com/UIGF-org/mihoyo-api-collect/blob/main/other/authentication.md#x-rpc-device_fp
x-rpc-device_fp只说了获取的url，没有具体说明获取方法，在哪里可以找到？

**Comment by BTMuli** [comment]
> https://github.com/UIGF-org/mihoyo-api-collect/blob/main/other/authentication.md#x-rpc-device_fp x-rpc-device_fp只说了获取的url，没有具体说明获取方法，在哪里可以找到？

[getDeviceFp.test.ts](https://github.com/BTMuli/TGAssistant/blob/master/test/getDeviceFP.test.ts)

**Comment by TRIM5829** [comment]
> > https://github.com/UIGF-org/mihoyo-api-collect/blob/main/other/authentication.md#x-rpc-device_fp x-rpc-device_fp只说了获取的url，没有具体说明获取方法，在哪里可以找到？
> 
> [getDeviceFp.test.ts](https://github.com/BTMuli/TGAssistant/blob/master/test/getDeviceFP.test.ts)

谢谢。正在测试通过扫码获取gametoken转stoken，但总是提示当前设备存在风险。我想应该是没有这个请求头吧（或者说还有其他鉴权方法？）
另外，这个用途如果不是很重要的话，随机生成一串device_fp，米会验证吗

**Comment by BTMuli** [comment]
> 谢谢。正在测试通过扫码获取gametoken转stoken，但总是提示当前设备存在风险。我想应该是没有这个请求头吧（或者说还有其他鉴权方法？） 另外，这个用途如果不是很重要的话，随机生成一串device_fp，米会验证吗

**米游社会验证生成 device_fp 的合法性**，所以要么使用现成的合法的数据 (`device_fp` + `device_id` + `device_seed_time`等)，要么生成后携带该 `device_fp` 作为新 header 重新获取，校验返回结果是否一致。

上面测试代码其实有点过时了，目前个人在用的是 [这段代码](https://github.com/BTMuli/TeyvatGuide/blob/master/src/web/request/getDeviceFp.ts)，相关调用在[这里](https://github.com/BTMuli/TeyvatGuide/blob/298096858fefb48167b47e1e3d698142d49ba896/src/pages/common/Config.vue#L252-L316)

**Comment by qhy040404** [comment]
> 我生成的DS问题,我把**&b={}**删除了,设置为空字符串后请求,遇到错误码1034了,请求遇到验证码怎么办,我看文档好像没有解决方法

GET方法没有body你传个{}那肯定对不上啊
